### DISCUSSION TITLE
KV engine version 2 - Delete a single secret from a path containing other secrets

### DISCUSSION THREAD

**Discussion by CruzanCaramele** [hvac/hvac#987] [Q&A] [answered]
Hello all,
I have key value secrets in a single path such as **my_engine/developers/teststage/secrets**

in this path I have a couple of key-value secrets such as:

**myfoo** : **myfoosecrete**
**mybar**:  **mybardsecret**
**myother** : **myothersecret**

I want to delete single secret such as **myfoo** from  **my_engine/developers/teststage/secrets/myfoo**

when i use the following:
secret_path = developers/teststage/secrets
mount_point = my_engine
```
self.client.secrets.kv.v2.delete_latest_version_of_secret(
                mount_point=mount_point, path=secret_path
            )
```
it deletes the entire path although previous versions are available.
However I only want to remove the single secret

**Comment by briantist** [comment, marked as answer]
Hi @CruzanCaramele ,

In Vault, a "secret" refers to the entire path, so it encompasses all of the key-value pairs within that path. To "delete" a single kv-pair is to write a new secret or secret version with all of the kv pairs intact except the one you want gone.

**Reply by CruzanCaramele** [reply]
@briantist  indeed , I did exactly what you said now:
- read the key-value pairs in the path
- remove the single key-value pair I want with Python's **del**
- write back the other key-value pairs to the same path which then creates a new version of the path or secret.

Thank you !!
